common/src/main/java/org/apache/apex/common/util/JarHelper.java

```diff
@@ -186,4 +190,62 @@ private void addEntry(File file, String name) throws IOException
       jos.closeEntry();
     }
   }
+
+  /**
+   * Returns a full path to the jar-file that contains the given class and all full paths to dependent jar-files
+   * that are defined in the property "apex-dependencies" of the manifest of the root jar-file
+   * @param classPath Class path
+   * @return List of jar-files
+   */
+  public static List<String> getApexDependentJarsByClass(String classPath)
+  {
+    try {
+      return getApexDependentJarsByClass(Thread.currentThread().getContextClassLoader().loadClass(classPath));
+    } catch (ClassNotFoundException ex) {
+      logger.error("Cannot find the class {}", classPath, ex);
+      throw new RuntimeException("Cannot find the class " + classPath, ex);
+    }
+  }
+
+  /**
+   * Returns a full path to the jar-file that contains the given class and all full paths to dependent jar-files
+   * that are defined in the property "apex-dependencies" of the manifest of the root jar-file
+   * @param clazz Class
+   * @return List of jar-files
+   */
+  public static List<String> getApexDependentJarsByClass(Class clazz)
+  {
+    List<String> list = new LinkedList<>();
+
+    try {
+      URLConnection conn = clazz.getResource(clazz.getSimpleName() + ".class").openConnection();
```

Comment:
There is no requirement not to create a jar file if getJar() is called for a class that is located on the local file system and does not belong to an existing jar file. I don't see why such requirement will exist and why for certain classes jar should be created on the fly and some classes require pre-existing jars. The behavior must be consistent across all classes that Apex loads.

Loading of the manifest file should be added to the getJar(). Ability to specify extra dependencies using "apex-dependencies" (or Class-Path) should not be limited to a specific jars or classes. It needs to be a generic feature available to all classes that Apex loads.

There is no need to make getJar() static. There is only one instance of JarHelper created.